### ISSUE TITLE
output cache not working?

### ISSUE THREAD

**Issue by fommil** [sbt/sbt-assembly#289]
I am trying to build on top of sbt-assembly with a plugin that does something like this

```
FileFunction.cached(streams.value.cacheDirectory / "my-plugin") { (in: Set[File]) =>
  stuff(in)
}(Set(assembly.value))
```

but it would appear that `assembly.value` **always** creates a new jar, regardless of the `cacheOutput` setting. i.e. the logic here appears to be broken as it is including something that is always changing https://github.com/sbt/sbt-assembly/blob/master/src/main/scala/sbtassembly/Assembly.scala#L85

**Comment by fommil** [comment]
seems I was on an older version, after upgrading to 0.14.6 the cache seems to work! w00t

**Comment by fommil** [comment]
Reopening because I find that I need to build the assembly jar TWICE before caching starts to kick in.

**Comment by fommil** [comment]
FYI my use of `assembly.value` is in https://github.com/sbt/sbt-proguard/issues/28

**Comment by xuwei-k** [comment]
ref https://github.com/sbt/sbt-assembly/issues/299. `sbt-test/caching/caching` test fail with sbt 1.x (success sbt 0.13.x)

**Comment by bpholt** [comment]
To confirm that https://github.com/sbt/util/pull/168 is the root cause of this issue, I built https://github.com/sbt/util/pull/169 locally and ran the `caching/caching` scripted test using the resulting sbt `1.2.0-SNAPSHOT`. With that build, the test passes, so once https://github.com/sbt/util/pull/169 is merged and released, we should be able to reenable `caching/caching`, closing this issue (and #299).

**Comment by limansky** [comment]
Hi! 

I've observed that this issue is still valid. Should I submit another issue or you can reopen this one?

Here is the description: https://github.com/sbt/sbt/issues/5567

The demonstration project is available at:
https://github.com/limansky/fatbug

Then run sbt clean publishLocal docker (you have to have docker running).
